test/AutoRest.TestServer.Tests/Mgmt/Unit/CodeModelExtensionTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using AutoRest.CSharp.Mgmt.Decorator;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests.Mgmt.Unit
+{
+    internal class CodeModelExtensionTests
+    {
+        [OneTimeSetUp]
+        public static void ApplyRenameRules()
+        {
+            var renameRules = new Dictionary<string, string>
+            {
+                { "Os", "OS" },
+                { "Ip", "IP" },
+                { "Ips", "IPs" },
+                { "ID", "Id" },
+                { "IDs", "Ids" },
+                { "VPN", "Vpn" },
+                { "WAN", "Wan" },
+                { "WANs", "Wans" },
+            };
+            CodeModelExtension.ApplyRenameRules(renameRules);
+        }
+
+        [TestCase("OsType", "OSType")]
+        [TestCase("DnsIp", "DnsIP")]
+        [TestCase("OsProfile", "OSProfile")]
+        [TestCase("ipAddress", "IPAddress")]
+        [TestCase("vipAddress", "VipAddress")]
+        [TestCase("publicIp", "PublicIP")]
+        [TestCase("publicIps", "PublicIPs")]
+        [TestCase("resourceId", "ResourceId")]
+        [TestCase("resourceIds", "ResourceIds")]
+        [TestCase("resourceIDs", "ResourceIds")]
+        [TestCase("resourceIdSuffix", "ResourceIdSuffix")]
+        [TestCase("IpsilateralDisablity", "IpsilateralDisablity")]
+        [TestCase("HumanIpsilateralDisablity", "HumanIpsilateralDisablity")]
+        [TestCase("DnsIpAddressIDsForWindowsOs", "DnsIPAddressIdsForWindowsOS")]
+        [TestCase("DnsIpAddressIDsForWindowsOsPlatform", "DnsIPAddressIdsForWindowsOSPlatform")]
+        [TestCase("something_IDs_Ip_Os", "Something_Ids_Ip_OS")]
+        [TestCase("Dynamic365IDs", "Dynamic365Ids")]
+        [TestCase("VirtualWAN", "VirtualWan")]
+        [TestCase("VirtualWANs", "VirtualWans")]
+        public void EnsureNameCaseTest(string name, string expected)
```

Comment:
Add a list of comments somewhere of use cases that we know this algorithm doesn't cover so we are explicitly calling them out.  for instance myvm we aren't covering since we are relying on Pascal / camel case to split the words.

Are we covering variable names here?  how do our helper methods to convert from variable to property names deal with names that start with one of these acronyms in camelCase
oSPlatform -> osPlatform
oSPlatform -> OSPlatform
vMThingie -> vmThingie
The rule should be that one cap all cap one small all small

Reply:
Unfortunately we are not covering variable names here. This is one of our known issues.
The name in the swagger might start with a upper case letter or a lower case letter. We are doing the transformation to variable names in [this function](https://github.com/Azure/autorest.csharp/blob/ea40037443519fdfd0bccbc4fc10918d90f46d0c/src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs#L76), but it is not actually looking for the words in a dictionary and will give you the result of `oSPlatform`. I already opened an issue to track that here: https://github.com/Azure/autorest.csharp/issues/1937
Considering all the things that go into public API start with an upper case letter, I did not take that into consideration in this PR.

Reply:
I think we will need to incorporate that here otherwise when we call ToVariableName in various places it will only ToLower the first character and not the full initialism

Reply:
We will do this in a separate PR